### PR TITLE
[Feat/LIVE-9350]:  send providerRedirectURL to swap live app

### PR DESCRIPTION
**Pull request by kallen-ledger**
### 📝 Description

* send `providerRedirectURL` to swap web app as part of state.

### ❓ Context

- **JIRA or GitHub issue**: 

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:**
  - 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Comment by gre** [comment]
-  npx changeset was attached.
-  Covered by automatic tests.
-  Impact of the changes:

please make sure to document these. if it can't be covered by tests, you can comment next to it why it can't / why it won't. 

🙏

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx [hunk 1]
```diff
@@ -163,6 +163,23 @@ const SwapForm = () => {
...
     const isCustomFee =
       feesStrategy === "slow" || feesStrategy === "fast" || feesStrategy === "custom";
+
+    const providerRedirectFromAccountId =
+      fromAccount &&
+      provider &&
+      walletApiPartnerList?.enabled &&
+      walletApiPartnerList?.params?.list.includes(provider)
+        ? accountToWalletAPIAccount(fromAccount, fromParentAccount)?.id
+        : fromAccount?.id;
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx:173]:
```suggestion
    const providerRedirectFromAccountId = accountToWalletAPIAccount(fromAccount, fromParentAccount)?.id;
```

Are we going to be always in wallet-api context? Do we really need this if/else?
(Inside getSwapWebAppInputProps, behind isSwapLiveAppEnabled)

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx [hunk 2]
```diff
@@ -174,6 +191,9 @@ const SwapForm = () => {
       customFeeConfig: customFeeConfig ? JSON.stringify(customFeeConfig) : undefined,
       providerType: exchangeRate?.providerType,
       cacheKey: v4(),
+      providerRedirectURL: `ledgerlive://discover/${getProviderName(
+        provider ?? "",
+      ).toLowerCase()}?${providerRedirectURLSearch.toString()}`,
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx:196]: Have we test this redirection with oneinch and paraswap?

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx (hunk 1 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx:173]: You are totally wright, I have another thing in mind.

**Review by sarneijim — APPROVED** [review]

**Pull request closed without merging**